packages/build/src/homebrew/generate-formula.ts

```diff
@@ -1,15 +1,33 @@
 import * as semver from 'semver';
 import type { GithubRepo } from '@mongodb-js/devtools-github-repo';
 
+/**
+ * When sending requests via Octokit, a situation can arise where the server closes the connection,
+ * but the client still believes it’s open and attempts to write to it,
+ * what leads to receiving an EPIPE error from the OS, indicating the connection has already been closed.
+ * In such cases, retrying the request can help establish a new, functional connection.
+ */
+async function getFileWithRetry(
+  homebrewCore: GithubRepo,
+  remainingRetries = 3
+) {
+  try {
+    return await homebrewCore.getFileContent('Formula/m/mongosh.rb', 'master');
+  } catch (error: any) {
+    if (error.message.includes('EPIPE') && remainingRetries > 0) {
+      return await getFileWithRetry(homebrewCore, remainingRetries - 1);
```

Comment:
Just a small suggestion, but I'd add `console.error(error)` in this branch so that it's a bit clearer whether there were any retries